artifactory.py

```diff
@@ -1015,14 +1036,50 @@ def del_properties(self, pathobj, props, recursive):
     def scandir(self, pathobj):
         return _ScandirIter((pathobj.joinpath(x) for x in self.listdir(pathobj)))
 
-    def writeto(self, fd, out, chunk_size):
-        url = str(fd)
-        res = fd.session.get(url, stream=True, verify=True, cert=None)
-        if res.status_code != 200:
-            raise RuntimeError(res.status_code)
-        for chunk in res.iter_content(chunk_size=chunk_size):
-            if chunk:
-                out.write(chunk)
+    def writeto(
+        self,
+        arti_cls,
+        output,
+        chunk_size=1024,
+        progress_func=print_download_progress,
+        *args,
+        **kwargs
+    ):
+        """
+        Downloads large file in chunks and prints progress
+        :param arti_cls: ArtifactoryPath class
+        :param output: file path of output file
+        :param chunk_size: chunk size in bytes, recommend. eg 1024*1024 is 1Mb
+        :param progress_func: custom function to print output, otherwise uses default print_download_progress
+        :param args: *args to supply for your custom progress_func
+        :param kwargs: **kargs to supply for your custom progress_func
+        :return: None
+        """
+        response = self.get_response(arti_cls)
+
+        bytes_read = 0
```

Comment:
A variable should be declared close to the its usage as much as possible. In this case before `for`
```python
bytes_read = 0
            for chunk in response.iter_content(chunk_size=chunk_size):
                bytes_read += len(chunk)
                if callable(progress_func):
                    progress_func(bytes_read, file_size, *args, **kwargs)
                file.write(chunk)
```